packages/yew/src/platform/fmt/buffer.rs

```diff
@@ -0,0 +1,212 @@
+use std::cell::UnsafeCell;
+use std::fmt::{self, Write};
+use std::marker::PhantomData;
+use std::rc::Rc;
+use std::task::{Poll, Waker};
+
+use futures::stream::{FusedStream, Stream};
+
+static BUF_SIZE: usize = 1024;
+
+enum BufStreamState {
+    Ready,
+    Pending(Waker),
+    Done,
+}
+
+struct Inner {
+    buf: String,
+    state: BufStreamState,
+
+    // This type is not send or sync.
+    _marker: PhantomData<Rc<()>>,
```

Comment:
Why not do:
```rust
impl !Send for Inner {}
impl !Sync for Inner {}
```

Reply:
This is due to that negative impls is not stable.

See: https://doc.rust-lang.org/unstable-book/language-features/negative-impls.html